tfhe/src/core_crypto/commons/math/decomposition/decomposer.rs

```diff
@@ -353,40 +390,96 @@ where
     /// use tfhe::core_crypto::commons::parameters::{
     ///     CiphertextModulus, DecompositionBaseLog, DecompositionLevelCount,
     /// };
+    ///
+    /// let decomposition_base_log = DecompositionBaseLog(4);
+    /// let decomposition_level_count = DecompositionLevelCount(3);
+    /// let cihertext_modulus = CiphertextModulus::try_new((1 << 64) - (1 << 32) + 1).unwrap();
+    ///
     /// let decomposer = SignedDecomposerNonNative::new(
-    ///     DecompositionBaseLog(4),
-    ///     DecompositionLevelCount(3),
-    ///     CiphertextModulus::try_new((1 << 64) - (1 << 32) + 1).unwrap(),
+    ///     decomposition_base_log,
+    ///     decomposition_level_count,
+    ///     cihertext_modulus,
     /// );
     ///
+    /// let basis = 2i64.pow(decomposition_base_log.0.try_into().unwrap());
+    /// let half_basis = basis / 2;
+    ///
     /// // These two values allow to take each arm of the half basis check below
```

Comment:
Not sure about this, the second does not seem to achieve this and I'm not sure where this number comes from. A number that does is 2^63 - 2^32 + 1 = 9223372032559808513. This has decomposition 0,0,8 (while 2^63 has 0,0,-8), I've just taken its negative.

Reply:
not sure I understand that ?

Reply:
It's related to my other remark just below this one in that the code comment claims to "allow to take each arm of the half basis" but the two values don't actually do this, there is no arm which gives a maximal positive digit (i.e. a digit of half_basis = 8, so changing 16982820785129133100u64 to (for example) 9223372032559808513u64 would achieve this as the latter decomposes as 0,0,8 so we do indeed get the maximal positive digit of 8 in the decomposition.
The current value 16982820785129133100u64 decomposes to -5,-4,-1 (so no 8).

Reply:
ah yes thanks